content/file/file.go

```diff
@@ -482,12 +486,15 @@ func (s *Store) pushDir(name, target string, expected ocispec.Descriptor, conten
 		return fmt.Errorf("failed to save gzip to %s: %w", gzPath, err)
 	}
 
-	checksum := expected.Annotations[AnnotationDigest]
 	buf := bufPool.Get().(*[]byte)
 	defer bufPool.Put(buf)
-	if err := extractTarGzip(target, name, gzPath, checksum, *buf); err != nil {
-		return fmt.Errorf("failed to extract tar to %s: %w", target, err)
+	if s.AutoExtract {
+		checksum := expected.Annotations[AnnotationDigest]
+		if err := extractTarGzip(target, name, gzPath, checksum, *buf); err != nil {
+			return fmt.Errorf("failed to extract tar to %s: %w", target, err)
+		}
 	}
+
```

Comment:
If we don't do auto unpack, we should save the gz file to the target path.

Reply:
Changed line 271 to address this.